### PR TITLE
Add IMESSCD

### PR DESCRIPTION
**Pull request by volllly**

### CODE REVIEW

File: src/fiskaltrust.ifPOS/v2/me/IMESSCD.cs [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+namespace fiskaltrust.ifPOS.v2.me
+{
+    [ServiceContract]
+    public interface IMESSCD
+    {
+        [OperationContract(Name = "v2/RegisterInvoice")]
+        Task<RegisterInvoiceResponse> RegisterInvoiceAsync(RegisterInvoiceRequest RegisterInvoiceRequest);
+
+        [OperationContract(Name = "v2/RegisterTCR")]
+        Task<RegisterTCRResponse> RegisterTCRAsync(RegisterTCRRequest RegisterTCRRequest);
```
Comment (TSchmiedlechner) [inline src/fiskaltrust.ifPOS/v2/me/IMESSCD.cs:12]:
```suggestion
        Task<RegisterTCRResponse> RegisterTCRAsync(RegisterTCRRequest registerTCRRequest);
```

File: src/fiskaltrust.ifPOS/v2/me/Models/ApprovalType.cs [hunk 2]
```diff
@@ -0,0 +1,14 @@
+using System.Runtime.Serialization;
+namespace fiskaltrust.ifPOS.v2.me
+{
+    [DataContract]
+    public class ApprovalType
+    {
+        public decimal? DiscountAmt { get; set; }
```
Comment (TSchmiedlechner) [inline src/fiskaltrust.ifPOS/v2/me/Models/ApprovalType.cs:7]: I think we'll need to add the `[DataMember(Order = xxx)]` attribute to all of these properties to make it work with grpc-net :(

File: src/fiskaltrust.ifPOS/v2/me/IMESSCD.cs [hunk 3]
```diff
@@ -0,0 +1,17 @@
+using System.ServiceModel;
+using System.Threading.Tasks;
+namespace fiskaltrust.ifPOS.v2.me
+{
+    [ServiceContract]
+    public interface IMESSCD
+    {
+        [OperationContract(Name = "v2/RegisterInvoice")]
+        Task<RegisterInvoiceResponse> RegisterInvoiceAsync(RegisterInvoiceRequest RegisterInvoiceRequest);
```
[the remark this answers is not in the document]
Comment (TSchmiedlechner) [reply src/fiskaltrust.ifPOS/v2/me/IMESSCD.cs]: Done 😃

File: src/fiskaltrust.ifPOS/v2/me/Models/ApprovalType.cs (hunk 2 above)
Comment (TSchmiedlechner) [reply src/fiskaltrust.ifPOS/v2/me/Models/ApprovalType.cs]: Done 😃

File: src/fiskaltrust.ifPOS/v1/me/IMESSCD.cs [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+        /// Registers an electronic cash device (TCR) at the central invoice register (CIS), and returns the TCR code that is used for further operations.
+        /// </summary>
+        [OperationContract(Name = "v2/RegisterTCR")]
+        Task<RegisterTcrResponse> RegisterTcrAsync(RegisterTcrRequest registerTCRRequest);
+
+        /// <summary>
+        /// Unregisters an electronic cash device (TCR) from the central invoice register (CIS).
+        /// </summary>
+        [OperationContract(Name = "v2/UnregisterTCR")]
+        Task UnregisterTcrAsync(RegisterTcrRequest registerTCRRequest);
```
Comment (volllly) [inline src/fiskaltrust.ifPOS/v1/me/IMESSCD.cs:22]: Maybe add a `UnregisterTcrRequest` to theinterface which is just
```cs
public class UnregisterTcrRequest : RegisterTcrAsync { }
```

File: src/fiskaltrust.ifPOS/v1/me/Models/RegisterCashDepositRequest.cs [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+        /// When not null, signalizes that this receipt was late-signed because of the given reason. Null when the receipt was not processed in late-signing mode.
+        /// </summary>
+        [DataMember(Order = 40)]
+        public SubsequentDeliveryType? SubsequentDeliveryType { get; set; }
+
+        /// <summary>
+        /// Amount that is deposited in the till.
+        /// </summary>
+        [DataMember(Order = 50)]
+        public decimal Amount { get; set; }
```
Comment (volllly) [inline src/fiskaltrust.ifPOS/v1/me/Models/RegisterCashDepositRequest.cs:37]: Maybe call this `AbsoluteAmount` to clarify that this should be the absolute amount cash in the register

File: src/fiskaltrust.ifPOS/v1/me/Models/RegisterCashWithdrawalRequest.cs [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+        /// When not null, signalizes that this receipt was late-signed because of the given reason. Null when the receipt was not processed in late-signing mode.
+        /// </summary>
+        [DataMember(Order = 50)]
+        public SubsequentDeliveryType? SubsequentDeliveryType { get; set; }
+
+        /// <summary>
+        /// Amount that is withdrawn from the till.
+        /// </summary>
+        [DataMember(Order = 60)]
+        public decimal Amount { get; set; }
```
Comment (volllly) [inline src/fiskaltrust.ifPOS/v1/me/Models/RegisterCashWithdrawalRequest.cs:37]: Maybe call this `RelativeAmount` to clarify that this should be the amount of cash taken from the register

File: src/fiskaltrust.ifPOS/v1/me/IMESSCD.cs [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+        /// Registers a cash deposit to the cash registers at the central invoice register (CIS).
+        /// </summary>
+        [OperationContract(Name = "v2/RegisterCashDeposit")]
+        Task<RegisterCashDepositResponse> RegisterCashDepositAsync(RegisterCashDepositRequest registerCashDepositRequest);
+
+        /// <summary>
+        /// Registers a cash withdrawal from the cash registers at the central invoice register (CIS).
+        /// </summary>
+        [OperationContract(Name = "v2/RegisterCashWithdrawal")]
+        Task<RegisterCashWithdrawalResponse> RegisterCashWithdrawalAsync(RegisterCashWithdrawalRequest registerCashDepositRequest);
```
Comment (volllly) [inline src/fiskaltrust.ifPOS/v1/me/IMESSCD.cs:34]:
```suggestion
        Task RegisterCashWithdrawalAsync(RegisterCashWithdrawalRequest registerCashDepositRequest);
```
The `RegisterCashWithdrawalResponse` has no properties so we can remove it?

File: src/fiskaltrust.ifPOS/v1/me/Models/InvoiceDetails.cs [hunk 8]
```diff
@@ -0,0 +1,147 @@
...
+        /// The total amount of goods and services delivered when VAT is not charged. Null if not applicable.
+        /// </summary>
+        [DataMember(Order = 110)]
+        public decimal? TaxFreeAmount { get; set; }
+
+        /// <summary>
+        /// Details about all payments made for this invoice.
+        /// </summary>
+        [DataMember(Order = 120)]
+        public List<InvoicePayment> PaymentDetails { get; set; }
```
Comment (volllly) [inline src/fiskaltrust.ifPOS/v1/me/Models/InvoiceDetails.cs:81]: Can we make this optional? `?`

File: src/fiskaltrust.ifPOS/v1/me/Models/InvoiceDetails.cs [hunk 9]
```diff
@@ -0,0 +1,147 @@
...
+
+        /// <summary>
+        /// Details about all items included in the invoice.
+        /// </summary>
+        [DataMember(Order = 130)]
+        public List<InvoiceItem> ItemDetails { get; set; }
```
Comment (volllly) [inline src/fiskaltrust.ifPOS/v1/me/Models/InvoiceDetails.cs:87]: Can we make this optional? `?`

File: src/fiskaltrust.ifPOS/v1/me/Models/InvoiceDetails.cs [hunk 10]
```diff
@@ -0,0 +1,147 @@
...
+
+        /// <summary>
+        /// Details about all special fees included in the invoice.
+        /// </summary>
+        [DataMember(Order = 140)]
+        public List<InvoiceFee>? Fees { get; set; }
```
Comment (volllly) [inline src/fiskaltrust.ifPOS/v1/me/Models/InvoiceDetails.cs:93]: Can we make this optional? `?`

File: azure-pipelines.yml [hunk 11]
```diff
@@ -32,12 +32,14 @@ jobs:
...
         semverVersion: 'v1'
-    
+        
     - task: DotNetCoreCLI@2
       displayName: Restore
       inputs:
         command: 'restore'
         projects: '**/*.csproj'
+        feedsToUse: 'select'
+        vstsFeed: 'dev'   # TODO Remove before merge
```
Comment (TSchmiedlechner) [inline azure-pipelines.yml:42]: Don't forget 😅

File: src/fiskaltrust.Middleware.Interface.Client.Common.StrongName/fiskaltrust.Middleware.Interface.Client.Common.StrongName.csproj [hunk 12]
```diff
@@ -9,7 +9,7 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="fiskaltrust.interface.StrongName" Version="1.3.1" />
+    <PackageReference Include="fiskaltrust.interface.StrongName" Version="1.3.1-ci-22188-54837" />
```
Comment (TSchmiedlechner) [inline src/fiskaltrust.Middleware.Interface.Client.Common.StrongName/fiskaltrust.Middleware.Interface.Client.Common.StrongName.csproj:12]: Don't forget to create a release version and update

File: src/fiskaltrust.Middleware.Interface.Client.Common/fiskaltrust.Middleware.Interface.Client.Common.csproj [hunk 13]
```diff
@@ -7,6 +7,6 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="fiskaltrust.interface" Version="1.3.1" />
+    <PackageReference Include="fiskaltrust.interface" Version="1.3.1-ci-22188-54837" />
```
Comment (TSchmiedlechner) [inline src/fiskaltrust.Middleware.Interface.Client.Common/fiskaltrust.Middleware.Interface.Client.Common.csproj:10]: Don't forget to create a release version and update

File: src/fiskaltrust.Middleware.Interface.Client.Grpc/fiskaltrust.Middleware.Interface.Client.Grpc.csproj [hunk 14]
```diff
@@ -5,7 +5,7 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="fiskaltrust.interface" Version="1.3.1" />
+    <PackageReference Include="fiskaltrust.interface" Version="1.3.1-ci-22188-54837" />
```
Comment (TSchmiedlechner) [inline src/fiskaltrust.Middleware.Interface.Client.Grpc/fiskaltrust.Middleware.Interface.Client.Grpc.csproj:8]: Don't forget to create a release version and update

File: src/fiskaltrust.Middleware.Interface.Client.Http.StrongName/fiskaltrust.Middleware.Interface.Client.Http.StrongName.csproj [hunk 15]
```diff
@@ -6,7 +6,7 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="fiskaltrust.interface.StrongName" Version="1.3.1" />
+    <PackageReference Include="fiskaltrust.interface.StrongName" Version="1.3.1-ci-22188-54837" />
```
Comment (TSchmiedlechner) [inline src/fiskaltrust.Middleware.Interface.Client.Http.StrongName/fiskaltrust.Middleware.Interface.Client.Http.StrongName.csproj:9]: Don't forget to create a release version and update

File: src/fiskaltrust.Middleware.Interface.Client.Http/fiskaltrust.Middleware.Interface.Client.Http.csproj [hunk 16]
```diff
@@ -6,7 +6,7 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="fiskaltrust.interface" Version="1.3.1" />
+    <PackageReference Include="fiskaltrust.interface" Version="1.3.1-ci-22188-54837" />
```
Comment (TSchmiedlechner) [inline src/fiskaltrust.Middleware.Interface.Client.Http/fiskaltrust.Middleware.Interface.Client.Http.csproj:9]: Don't forget to create a release version and update

File: src/fiskaltrust.Middleware.Interface.Client.Soap/fiskaltrust.Middleware.Interface.Client.Soap.csproj [hunk 17]
```diff
@@ -6,7 +6,7 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="fiskaltrust.interface" Version="1.3.1" />
+	  <PackageReference Include="fiskaltrust.interface" Version="1.3.1-ci-22188-54837" />
```
Comment (TSchmiedlechner) [inline src/fiskaltrust.Middleware.Interface.Client.Soap/fiskaltrust.Middleware.Interface.Client.Soap.csproj:9]: Don't forget to create a release version and update

File: src/fiskaltrust.ifPOS/v1/me/Models/RegisterCashDepositRequest.cs (hunk 5 above)
Comment (TSchmiedlechner) [reply src/fiskaltrust.ifPOS/v1/me/Models/RegisterCashDepositRequest.cs:37]: So, this is actually the amount of cash that is available in the cash register after this operation, right?
So e.g. if I already had 10€, and I'm adding 90€ more, this value would be 100€?

**Review by TSchmiedlechner — APPROVED** [review]

**Pull request merged**